src/Request/CurrentDateTimeConverter.php

```diff
@@ -0,0 +1,56 @@
+<?php
+declare(strict_types=1);
+
+namespace LoyaltyCorp\RequestHandlers\Request;
+
+use DateTime;
+use DateTimeInterface;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter\ParamConverterInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Create a date time on demand in controllers.
+ */
+class CurrentDateTimeConverter implements ParamConverterInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function apply(Request $request, ParamConverter $configuration)
+    {
+        $param = $configuration->getName();
+
+        // Don't override existing values.
+        if ($request->attributes->has($param) === true &&
+            $request->attributes->get($param) !== null) {
```

Comment:
If we end up with a string in the param name, you want to let it past unmodified?

I would check for instanceof DateTime or a string?

Reply:
In this case if we end up with an attribute that matches (which is a string), but the method is typed as a DateTime with the paramconverter annotation we might get a fatal

Reply:
Yeah, good call. I'll change that.

Reply:
Actually, let me back that up - while that's potentially true - it's not the responsibility of this class to validate data, that's should be done in the Validator / Request objects.